### PR TITLE
Add Linux useland script, kernel module and doc

### PR DESCRIPTION
**Pull request by conchyliculture**
We add:
- A simple kernel module to access fan PWM via hwmon
- A simple userland script that will try to maintain GPU temperature (no need for above module)
- Docs

### CODE REVIEW

File: linux/asus_fc2_module/asus_fc2.c [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+#include <linux/init.h>
+#include <linux/kernel.h> // for KERN_INFO
+#include <linux/module.h>
+#include <linux/regmap.h>
+
+#define ASUS_FC2_VERSION "0.0.20200412"
+
+#define ASUS_FC2_DEVICE_ADDR 0x2a
+#define ASUS_FC2_DEVICE_ADDR_BACKUP                                            \
+  0x68 // Is apparently exactly the same as 0x2a
```
Comment (x0rbl) [inline linux/asus_fc2_module/asus_fc2.c]: Put this all on one line

File: linux/asus_fc2_module/asus_fc2.c [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+static int asus_fc2_probe(struct i2c_client *client,
+                          const struct i2c_device_id *id) {
+  struct device *dev = &client->dev;
+  struct device *hwmon_dev;
+  struct regmap *regmap;
+  unsigned int regval_low;
+  unsigned int regval_high;
+  int ret;
+
+  dev = &client->dev;
```
Comment (x0rbl) [inline linux/asus_fc2_module/asus_fc2.c]: Duplicate of line 115; you can remove this

File: linux/asus_fc2_module/asus_fc2.c [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  case hwmon_pwm:
+    switch (attr) {
+    case hwmon_pwm_input:
+      return 0644;
+    }
+    break;
+  default:
+    break;
+  }
+  return 0;
```
Comment (x0rbl) [inline linux/asus_fc2_module/asus_fc2.c]: The double switch is a little confusing.  Feel free to disregard if this is the preferred style (I notice switches are used everywhere else).  But probably the most readable way to write it is:

```
if (type == hwmon_pwm && attr == hwmon_pwm_input) {
  return 0644;
}
return 0;
```

**Review by x0rbl — CHANGES_REQUESTED** [review]

File: linux/asus_fc2_module/asus_fc2.c (hunk 1 above)
Comment (conchyliculture) [reply linux/asus_fc2_module/asus_fc2.c]: who are you to defy clang-format? :3

File: linux/asus_fc2_module/asus_fc2.c (hunk 3 above)
Comment (conchyliculture) [reply linux/asus_fc2_module/asus_fc2.c]: oh it's just because here I ended up with very few options (only type=pwm & attr = pwm_input)

**Review by x0rbl — APPROVED** [review]

**Pull request merged**